libyul/AsmParser.cpp

```diff
@@ -779,7 +779,7 @@ void Parser::checkBreakContinuePosition(std::string const& _which)
 	}
 }
 
-bool Parser::isValidNumberLiteral(std::string const& _literal)
+bool Parser::isValidNumberLiteral(std::string_view const _literal)
```

Comment:
```suggestion
bool Parser::isValidNumberLiteral(std::string_view _literal)
```

Reply:
I think the `const` is perfectly fine here. It expresses the intent that "hey, this string view is not going to mutate its contents" - which is something I would expect from a method that checks whether something is a valid number literal.

Reply:
But isn't string view already read-only anyway? I thought that const wouldn't be necessary here. Also, the header should be adjusted to match the signature then, no?

Reply:
It is read-only on the string but it can be mutated. For example:

```cpp
std::string s = "abc";
std::string_view sv = s;
sv.remove_prefix(1);
std::cout << sv << '\n' // prints "bc"
```

So my comment wasn't quite correct as in the contents (ie the referred-to string) are indeed immutable but you can still mutate the stringview itself. Unless it's const. :)

Regarding the signature: The const in the signature is superfluous as it is by-value (ie the object is copied into the the function). There is no difference. Try `void f(int) {}` and `void f(int const) {}`. The compiler will complain about  a redefinition. In the implementation, though, we are acting on a reference to that copy. There it can make sense to mark it const.